File: codember/2023/CHALLENGE_02/ts/main.ts
function miniCompiler(program: string): string {
  let numericValue: number = 0;
  let result: string = "";

  for (let operation of program) {
    switch (operation) {
      case "#":
        numericValue += 1;
        break;
      case "@":
        numericValue -= 1;
        break;
      case "*":
        numericValue *= numericValue;
        break;
      case "&":
        result += numericValue.toString();
        break;
    }
  }

  return result;
}

const input: string =
  "&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&";

console.log(miniCompiler(input));

File: codember/2023/CHALLENGE_02/rust/src/main.rs
fn mini_compiler(program: &str) -> String {
    let mut value = 0;
    let mut output = String::new();

    for symbol in program.chars() {
        match symbol {
            '#' => value += 1,
            '@' => value -= 1,
            '*' => value *= value,
            '&' => output.push_str(&value.to_string()),
            _ => (),
        }
    }

    output
}

fn main() {
    let input = "&###@&*&###@@##@##&######@@#####@#@#@#@##@@@@@@@@@@@@@@@*&&@@@@@@@@@####@@@@@@@@@#########&#&##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@##@@&";
    let output = mini_compiler(input);
    println!("submit {}", output);
}
